src/js/worklets/profile_header.js

```diff
@@ -0,0 +1,13 @@
+import { useAnimatedStyle, withTiming } from 'react-native-reanimated';
+
+export function profileHeaderAnimation(scrollY, threshold, topBarHeight) {
+  return useAnimatedStyle(() => {
+    const opacity = scrollY.value < threshold ? withTiming(0) : withTiming(1);
+    const translateY = scrollY.value < threshold ? withTiming(topBarHeight) : withTiming(0);
+
+    return {
+      opacity: opacity,
+      transform: [{ translateY: translateY }],
+    };
+  });
+}
```

Comment:
Does this need to be defined in JS because it runs in the animation thread?

Reply:
yes i forgot to add `worklet`, just added

Reply:
I believe this looks cleaner @mohsen-ghafouri ! :)

btw,  we don't need that `worklet` directive anymore for the animated hooks, reanimated automatically converts them even if not present, but only when it detects an animated hook is being used.